samples/meta-space-warp-sample/main.gd

```diff
@@ -0,0 +1,70 @@
+extends StartXR
+
+const TURN_THRESHOLD := 0.8
+const TURN_ANGLE := TAU / 16
+const MOVE_SPEED := 2.0
+
+var movement_input := Vector2.ZERO
+var countdown_to_check_space_warp_enabled: int = 3
+
+@onready var xr_origin: XROrigin3D = $XROrigin3D
+@onready var xr_camera: XRCamera3D = $XROrigin3D/XRCamera3D
+@onready var right_controller_label: Label3D = $XROrigin3D/RightHand/Label3D
+@onready var turn_timer: Timer = $XROrigin3D/TurnTimer
+
+func _process(delta: float) -> void:
+	if countdown_to_check_space_warp_enabled > 0:
+		countdown_to_check_space_warp_enabled -= 1
+		if countdown_to_check_space_warp_enabled == 0:
+			var fb_space_warp = Engine.get_singleton("OpenXRFbSpaceWarpExtensionWrapper")
+			if not fb_space_warp.is_enabled():
+				right_controller_label.text = right_controller_label.text.replace("ENABLED", "DISABLED")
+
+
+func _physics_process(delta: float) -> void:
+	if movement_input != Vector2.ZERO:
+		xr_origin.position.z += -movement_input.y * delta * MOVE_SPEED
+		xr_origin.position.x += movement_input.x * delta * MOVE_SPEED
+
+
+func _on_right_hand_button_pressed(name: String) -> void:
+	if name == "ax_button":
+		var fb_space_warp = Engine.get_singleton("OpenXRFbSpaceWarpExtensionWrapper")
+		fb_space_warp.set_space_warp_enabled(!fb_space_warp.is_enabled())
+
+		if fb_space_warp.is_enabled():
+			right_controller_label.text = right_controller_label.text.replace("DISABLED", "ENABLED")
+		else:
+			right_controller_label.text = right_controller_label.text.replace("ENABLED", "DISABLED")
+
+
+func check_turn(name: String, value: Vector2) -> void:
+	if not turn_timer.is_stopped():
+		return
+
+	if name == "primary":
+		if value.x > TURN_THRESHOLD:
+			rotate_player(TURN_ANGLE)
+			turn_timer.start()
+		if value.x < -TURN_THRESHOLD:
+			rotate_player(-TURN_ANGLE)
+			turn_timer.start()
```

Comment:
We should be calling `skip_space_warp_frame()` when we do snap turns.

I'm not actually able to see any artifacts from not doing it, but theoretically, because we're including the rotation since the last frame in the `appSpaceDeltaPose`, we should see it generating an in-between "smear" frame (which is not what a developer would want on snap turn, and why we'd skip space warp).

Because I'm not seeing the expected artifact, I'm worried we might not be doing the rotation in `appSpaceDeltaPose` correctly, which is another reason why I think we should have smooth turn in the sample to confirm.

Reply:
After discussing with Logan, and doing some experimentation, I've realized that I had incorrect ideas of what artifact we should (or should not) see here.

I think the current calculation of `appSpaceDeltaPose` is correct. However, I think it's still worth skipping AppSW on snap turns, and adding smooth turn to demonstrate the correctness of the calculation (which I was already seeing with snap turns not generating an artifact, but we won't any more if we skip AppSW on snap turns).